thoth/lab/inspection.py

```diff
@@ -457,13 +457,210 @@ def show_categories(inspection_df: pd.DataFrame):
     """List categories in the given DataFrame."""
     index = inspection_df.index.droplevel(-1).unique()
 
+    results_categories = {}
     for n, idx in enumerate(index.values):
-        print("\nCategory {}/{}".format(n + 1, len(index)))
+        print("\nClass {}/{}".format(n + 1, len(index)))
+
+        class_results = {}
         if len(index.names) > 1:
             for name, ind in zip(index.names, idx):
                 print(f"{name} :", ind)
+                class_results[name] = ind
         else:
             print(f"{index.names[0]} :", idx)
+            class_results[index.names[0]] = idx
+        results_categories[n + 1] = class_results
 
         frame = inspection_df.loc[idx]
         print("Number of rows (jobs) is:", frame.shape[0])
+
+    return results_categories
+
+
+def filter_inspection_list(ids_list: List[str], identifier_list: List[str]) -> dict:
+    """Filter inspection ids list according to the identifier selected."""
+    filtered_list_ids = {}
+
+    for identifier in identifier_list:
+        filtered_list_ids[identifier] = []
+
+    for ids in ids_list:
+        inspection_filter = "-".join(ids.split("-")[1:(len(ids.split("-")) - 1)])
+
+        if inspection_filter:
+            if inspection_filter in identifier_list:
+                filtered_list_ids[inspection_filter].append(ids)
+
+    tot_inspections_selected = sum([len(batch_n) for batch_n in filtered_list_ids.values()])
+    inspection_batches = [(batch_name, len(batch_count)) for batch_name, batch_count in filtered_list_ids.items()]
+    print(f"There are {tot_inspections_selected} inspection runs!: {inspection_batches} respectively")
```

Comment:
It would be good to use logging for messages (in general). This is a library which could be used in some CLI which would produce results. Printing stuff to stdout can be annoying in that case (and it would be good to turn these messages optionally of - e.g. logging's loglevel).